src/services/tokenService.ts

```diff
@@ -0,0 +1,17 @@
+import jwt from 'jsonwebtoken';
+import { User } from '../models/User';
+import { generalConfig } from '../config/generalConfig';
+
+// Generate a JWT Token for the given User
+export const generateToken = (user: User): string => {
+  return jwt.sign(
+    { id: user.id, email: user.email, role: user.role.toString },
```

Comment:
The sign in should be fetching that info from the DB when we call `user.login()`. 

I will update the `user.register()` to return the user instead of a boolean so that its available when the token is generated

Reply:
Ok thanks.